encoding/rs/gpu/extend_poly.go

```diff
@@ -0,0 +1,39 @@
+package gpu
+
+import (
+	"sync"
+
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/fft"
+	"github.com/Layr-Labs/eigenda/encoding/utils/gpu_utils"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/ingonyama-zk/icicle/v2/wrappers/golang/core"
+	bn254_icicle "github.com/ingonyama-zk/icicle/v2/wrappers/golang/curves/bn254"
+	"github.com/ingonyama-zk/icicle/v2/wrappers/golang/curves/bn254/ntt"
+)
+
+type GpuComputeDevice struct {
+	Fs *fft.FFTSettings
+	encoding.EncodingParams
+	NttCfg  core.NTTConfig[[bn254_icicle.SCALAR_LIMBS]uint32]
+	GpuLock *sync.Mutex
+}
+
+// Encoding Reed Solomon using FFT
+func (g *GpuComputeDevice) ExtendPolyEval(coeffs []fr.Element) ([]fr.Element, error) {
+
+	g.GpuLock.Lock()
+	defer g.GpuLock.Unlock()
+
+	scalarsSF := gpu_utils.ConvertFrToScalarFieldsBytes(coeffs)
```

Comment:
is it important that coeffs is not tampered with? Otherwise you could potentially convert it inplace to save memory. Not sure how much of an improvement it would be but could be interesting to benchmark. You're doing a lot of memory allocations + copies everywhere.